### PR TITLE
Auto value

### PR DESCRIPTION
**Pull request by piotr-sumo**

### CODE REVIEW

**Comment by piotr-sumo** [comment]
@trask @iNikem These are all classes that I found worth converting to AutoValue.

File: instrumentation/aws-sdk/aws-sdk-1.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/awssdk/v1_11/AwsClientInstrumentation.java [hunk 1]
```diff
@@ -54,7 +54,8 @@ public static void addHandler(
       if (!hasAgentHandler) {
         handlers.add(
             new TracingRequestHandler(
-                InstrumentationContext.get(AmazonWebServiceRequest.class, RequestMeta.class)));
+                InstrumentationContext.get(
```
Comment (anuraaga) [inline instrumentation/aws-sdk/aws-sdk-1.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/awssdk/v1_11/AwsClientInstrumentation.java:57]: Thanks for the help @piotr-sumo - in #2508 I will remove the `RequestMeta` class completely. Do you mind reverting the changes to the `awssdk` in this PR?

Comment (piotr-sumo) [reply instrumentation/aws-sdk/aws-sdk-1.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/awssdk/v1_11/AwsClientInstrumentation.java:57]: @anuraaga sure, no problem. I will revert those changes.

**Review by iNikem — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]

**Review by trask — APPROVED** [review]
Thanks @piotr-sumo!

**Comment by trask** [comment]
@piotr-sumo when you have time, can you merge `main` into your branch and resolve the merge conflicts?

**Pull request merged**